src/main/java/org/spongepowered/api/map/color/MapPalette.java

```diff
@@ -0,0 +1,143 @@
+/*
+ * This file is part of SpongeAPI, licensed under the MIT License (MIT).
+ *
+ * Copyright (c) SpongePowered <https://www.spongepowered.org>
+ * Copyright (c) contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.api.map.color;
+
+import com.google.common.collect.ImmutableCollection;
+import org.spongepowered.api.map.util.MapPaletteFactory;
+import org.spongepowered.api.util.Color;
+import org.spongepowered.api.util.generator.dummy.DummyObjectProvider;
+
+/**
+ * A representation of the palette of a map. This provides a set of constants
+ * that hold the named based colors in a map, along with methods to retrieve
+ * the full listing of all the colors and their shades.
+ */
+public final class MapPalette {
+
+    private static final MapPaletteFactory factory = null;
+
+    // SORTFIELDS:ON
+
+    public static final MapColor AIR = DummyObjectProvider.createFor(MapColor.class, "AIR");
```

Comment:
It's a singleton, there are more of them than are named, any method taking a color and a shade will return the cached (or lazy-loaded) shaded version of that color's unshaded entry and the new shade.